### PR TITLE
add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jntullo**
Add general guidelines that link to the api contributor’s guide blogpost.

@miq-bot add_label developer

### CODE REVIEW

**Comment by imtayadeway** [comment]
@jntullo I ❤️ this PR! I was just thinking, WDYT about copying/pasting the guide here instead of linking to it? I'd expect the `CONTRIBUTING.md` guide to be more of a living document that we may need to change over time, whereas the blogpost will reflect more the state of things on the date of publication. No strong feelings though - we can always revisit this in the event that there is something that needs to be updated, and I welcome other thoughts on this.

**Comment by abellotti** [comment]
Agreeing with @imtayadeway , the blogpost can reference the CONTRIBUTING.md file here, as this will change in this repo over time as we enhance the API.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+              :identifier: coffee_delete
+            :delete:
+            - :name: delete
+              :identifier: coffee_delete
+```
+
+
+Below is a breakdown of what some of the above elements mean:
+* **:coffees:**
+    * Defines the collection or subcollection portion of the URI in conjunction with the name and supported verbs (in our case, `GET api/coffee`, `POST /api/coffee`, etc..)
```
Comment (abellotti) [inline CONTRIBUTING.md]: above should be GET /api/coffees, POST /api/coffees

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+* **:verbs:**
+    * Specifies the [verb set](https://github.com/ManageIQ/manageiq-api/blob/4811ec9fcea2424a585560a761e05d56c53535f8/config/api.yml#L18-L36) that can be used on the collection
+* **:klass:** 
+    * Specifies the base class name of the resources under this collection 
+* **:name:**
+    * Name of the action to be performed 
+* **:identifier:**
+    * Name of the corresponding [product feature](https://github.com/ManageIQ/manageiq/blob/f77c75b4659f97be95f8976ef58abe521085c963/db/fixtures/miq_product_features.yml), which is used to control permissions to resources
+* **:collection_actions:** vs **:resource_actions:**
+    * Collection actions correspond to those against the collection `/api/coffee` and resource actions are those corresponding to a single resource `/api/coffee/:c_id`
```
Comment (abellotti) [inline CONTRIBUTING.md]: /api/coffees/... here and below

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+## Creating a PR
+Small PRs are always easier to review, and this example would be broken down into two PRs - the Coffee CRUD PR and a Donuts subcollection PR. Examples in the PR description help reviewers to understand what was added, and also helps the users to get up to speed quickly! 
+Here is a snippet of what we might include as an example for ordering a coffee:
+```
+POST /api/coffee/:id
+{
+    "action": "order",
+    "cream": true,
+    "sugar": false
+}
```
Comment (abellotti) [inline CONTRIBUTING.md]: I know we support above, would prefer primary form here below and leave above when we add support for POST /api/coffees/:id/order.

```
POST /api/coffees/:id
{
  "action" : "order",
  "resource" : {
    "cream" : true,
    "sugar" : false
  }
}
```

**Comment by abellotti** [comment]
Thanks @jntullo for updating the doc.

**Pull request merged**